src/org/exist/xquery/functions/array/ArrayFunction.java

```diff
@@ -248,8 +248,13 @@ public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathExce
                     case SIZE:
                         return new IntegerValue(array.getSize());
                     case GET:
-                        final IntegerValue index = (IntegerValue) args[1].itemAt(0);
-                        return array.get(index.getInt() - 1);
+                        final int index = ((IntegerValue) args[1].itemAt(0)).getInt();
+
+                        if (index <= 0 || index > array.getSize()) {
```

Comment:
@dizzzz this is not needed? don't confuse passing an `int` or an `IntegerValue` to `ArrayType`, it has two different `get` methods, and the IntegerValue one will cope with the less than zero position by raising the correct error.